modules/diskq/qdisk.c

```diff
@@ -342,31 +342,17 @@ qdisk_push_tail(QDisk *self, GString *record)
       self->hdr->write_head = QDISK_RESERVED_SPACE;
     }
 
-  if (!qdisk_is_space_avail(self, record->len))
+  if (!qdisk_is_space_avail(self, record->len - sizeof(guint32)))
```

Comment:
The responsibility of adding the length of the string was removed from `qdisk_push_tail`, so it should be removed from `qdisk_is_space_avail` as well, I mean it is hard to argue why I have to subtract that amount, and for sure remembering it is unlikely.

Reply:
Good point, I made your suggested adjustment.